pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -300,9 +364,29 @@ public boolean index(GenericRow row) {
     for (String metric : dataSchema.getMetricNames()) {
       FixedByteSingleColumnSingleValueReaderWriter readerWriter =
           (FixedByteSingleColumnSingleValueReaderWriter) columnIndexReaderWriterMap.get(metric);
-      int dicId = dictionaryMap.get(metric).indexOf(row.getValue(metric));
-      readerWriter.setInt(docId, dicId);
-      rawRowToDicIdMap.put(metric, dicId);
+      if (isNoDictionaryFixedWidthColumn(metric)) {
+        switch (dataSchema.getFieldSpecFor(metric).getDataType()) {
+          case SHORT:
```

Comment:
Not sure, but I am removing it for now, since I didn't add support for SHORT elsewhere. I think I saw SHORT in some plasces when I starrtarted off, and then not in the query path, so I will just include the 4 types here.